tidy3d/components/simulation.py

```diff
@@ -2287,9 +2290,23 @@ def get_neighboring_media(
             new_bounds = (center - dls[0] / 2, center + dls[1] / 2)
             new_geometry = set_bounds(structure.geometry, bounds=new_bounds, axis=axis)
 
+            # check if a regular medium is overlaying a 2D medium
+            media = self.intersecting_media(new_geometry, self.structures[istruct:])
+            if not all(isinstance(medium, Medium2D) for medium in media):
+                raise SetupError(
```

Comment:
maybe report the `istruct` of the structures. so the user knows which one is overlapping.